sdk/go/common/util/env/env.go

```diff
@@ -188,20 +218,57 @@ type value struct {
 	store    Store
 }
 
+func (v value) Validate() ValidateError {
```

Comment:
I don't think this changed with this PR, but it seems like `Validate` does not get run before we access via `Value()`. Did I just miss something? Or is that intentional?

Reply:
It gets called when `pulumi env` is run. This is intentional. The idea was to not people who were setting invalid vars that were not used.

As a follow up to this PR, we should probably require that some env vars are validated before proceeding.